src/js/select2/selection/search.js

```diff
@@ -194,11 +194,23 @@ define([
   };
 
   Search.prototype.searchRemoveChoice = function (decorated, item) {
+    var searchValue = item.text;
     this.trigger('unselect', {
       data: item
     });
 
-    this.$search.val(item.text);
+    if (
+      this.options.options.searchRemoveAttribute ||
+      typeof this.options.options.searchRemoveAttribute === 'string'
+    ) {
+      if (item[this.options.options.searchRemoveAttribute]) {
+        searchValue = item[this.options.options.searchRemoveAttribute];
+      } else {
+        searchValue = '';
```

Comment:
Looking at this again I should probably update this to set searchValue to the value of searchRemoveAttribute.. This would make it clearer to use an empty string to do nothing and also allow backspace to always insert a given string (not that I have any idea why you would want that)